Frameworks/QuartzCore/CALayer.mm

```diff
@@ -577,7 +577,10 @@ - (void)display {
                         ++tries;
                         continue;
                     default: {
-                        NSTraceError(TAG, @"Error %@ rendering %@.", renderError.get(), self);
+                        FAIL_FAST_MSG("Failed to render <%hs %p>: %hs",
```

Comment:
The error description can be augmented to include the failing HRESULT, printing it plus the layer should allow us to correlate events and layer classes.